core/src/main/java/com/cprcrack/videowebview/VideoEnabledWebView.kt

```diff
@@ -0,0 +1,132 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2019 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+package com.cprcrack.videowebview
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.os.Handler
+import android.os.Looper
+import android.util.AttributeSet
+import android.util.Log
+import android.webkit.WebChromeClient
+import android.webkit.WebView
+
+/**
+ * This class serves as a WebView to be used in conjunction with a VideoEnabledWebChromeClient.
+ * It makes possible:
+ * - To detect the HTML5 video ended event so that the VideoEnabledWebChromeClient can exit
+ * full-screen.
+ *
+ * Important notes:
+ * - Javascript is enabled by default and must not be disabled with getSettings().setJavaScriptEnabled(false).
+ * - setWebChromeClient() must be called before any loadData(), loadDataWithBaseURL() or loadUrl()
+ * method.
+ *
+ * @author Cristian Perez (http://cpr.name)
+ */
+open class VideoEnabledWebView : WebView {
+  inner class JavascriptInterface {
+    @android.webkit.JavascriptInterface @SuppressWarnings("unused")
+    fun notifyVideoEnd() {
+      // Must match Javascript interface method of VideoEnabledWebChromeClient
+      Log.d("___", "GOT IT")
+      // This code is not executed in the UI thread, so we must force that to happen
+      Handler(Looper.getMainLooper()).post {
+        videoEnabledWebChromeClient?.onHideCustomView()
+      }
+    }
+  }
+
+  private var videoEnabledWebChromeClient: VideoEnabledWebChromeClient? = null
+  private var addedJavascriptInterface: Boolean
+
+  @SuppressWarnings("unused")
+  constructor(context: Context?) : super(context!!) {
```

Comment:
We should take non-nullable `Context` in order to avoid the double bang operator.

Reply:
done